File: src/main/java/eu/mjindra/units/Length.java
package eu.mjindra.units;

/**
 * An enum representing length in DND.
 * @author Martin Jindra
 * @version 27.05.2023
 */
public enum Length implements Base {
    FEET(1),
    INCH(12),
    METER(0.3048F),
    CENTIMETER(30.48F);

    public final float val;

    Length(float val) {
        this.val = val;
    }
}

File: lib/dnd/unit/length.dart
import 'package:helios/dnd/unit/unit.dart';

/// An enum representing length in DND.
enum Length implements Unit {
  feet(1),
  inch(12),
  meter(0.3048),
  centimeter(30.48);

  const Length(this.val);

  final double val;
}
